### PR TITLE
Support for conditional compilation depending on target

### PR DESCRIPTION
**Pull request by mstade**
## Description

This PR implements support for conditional compiling of JavaScript features, depending on the target environment. The builder covers the same feature set as before, i.e. ratified standards by default, stages can still be enabled via flags.

This removes the `es2017` flag, since it's now a ratified standard. Those features are enabled by default. This PR should also remove any need for `es2017` style flags in the future, since the preset we use to build things essentially include the same as latest and thus those features would be enabled based on target environments instead.

## Motivation and Context

By using conditional compilation based on targets, it's easier to ensure we're only compiling what needs to be compiled, as opposed to taking least common denominator approach of always compiling anything back to es2015. Hopefully, this should improve the quality of the output, and over time provide performance improvements as well.

This means it's more difficult to guarantee consistent builds, since a query like `last 3 versions` for browser targets won't remain the same over time. Hopefully this is still not particularly tricky to reason about however.

## How Was This Tested?

All current tests pass, and a few tests were added to cover the new options. However, the test suite probably needs expansion.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change follows the style of this project
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have read the [contribution guidelines](../CONTRIBUTING.md)
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -146,24 +146,61 @@ If `NODE_ENV` is not set when ez-build is invoked in production mode, it will be
...
 *Note: the `NODE_ENV=production ez-build` syntax may not work on all systems. For compatibility across operating systems, you may want to consider using something like [cross-env](https://www.npmjs.com/package/cross-env).*
 
+### `--target-browsers <spec|false>`
+
+Define which browsers that ez-build should do its best to target when producing its output. What this means is that only those features that actually need compiling will be compiled, and the rest will be left alone. The `spec` string can be any query supported by [browserslist](https://github.com/ai/browserslist). This option defaults to `"last 3 versions"` which is a rather broad range, intended to capture most reasonably modern browsers.
+
+**Please note:** because of the rather freeform nature of target queries, it's very likely that you will have to quote the query, otherwise it won't be passed correctly. For instance, this will not be recognized properly:
+
+```bash
+$ ez-build --target-browsers last 3 versions
```
Comment (Poetro) [inline README.md]: Don't put bad examples at the top, as people will just copy paste the first one with the thought of "tl;dr".

Comment (mstade) [inline README.md]: Very good point, thanks buddy!

Comment (mstade) [inline README.md]: Fixed in 9e2f3d4.

**Comment by mstade** [comment]
@poetro does this look alright to you now?

**Pull request merged**